chains/evm/contracts/poolsV2/TokenPool.sol

```diff
@@ -43,10 +87,180 @@ abstract contract TokenPool is IPoolV2, TokenPoolV1 {
+      token: address(i_token),
+      sender: msg.sender,
+      amount: destTokenAmount
+    });
+
+    return (
+      Pool.LockOrBurnOutV1({
+        destTokenAddress: getRemoteToken(lockOrBurnIn.remoteChainSelector),
+        destPoolData: _encodeLocalDecimals()
+      }),
+      destTokenAmount
+    );
+  }
+
+  function releaseOrMint(
+    Pool.ReleaseOrMintInV1 calldata releaseOrMintIn,
+    uint16 finality
+  ) public virtual override(IPoolV2) returns (Pool.ReleaseOrMintOutV1 memory) {
+    uint256 localAmount = _calculateLocalAmount(
+      releaseOrMintIn.sourceDenominatedAmount, _parseRemoteDecimals(releaseOrMintIn.sourcePoolData)
+    );
+
+    _validateReleaseOrMint(releaseOrMintIn, localAmount, finality);
+
+    _releaseOrMint(releaseOrMintIn.receiver, localAmount);
+
+    emit ReleasedOrMinted({
+      remoteChainSelector: releaseOrMintIn.remoteChainSelector,
+      token: address(i_token),
+      sender: msg.sender,
+      recipient: releaseOrMintIn.receiver,
+      amount: localAmount
+    });
+
+    return Pool.ReleaseOrMintOutV1({destinationAmount: localAmount});
+  }
+
+  // ================================================================
+  // │                         Validation                           │
+  // ================================================================
+
+  /// @notice Validates the lock or burn request, applies any fast-finality fee, and enforces rate limits.
+  /// @dev The validation covers token support, RMN curse status, allowlist membership, onRamp access, and
+  /// rate limiting for both standard and fast-transfer lanes.
+  /// @param lockOrBurnIn The input to validate. Must reference a supported token, onRamp, and remote chain.
+  /// @param finality The finality depth requested by the message. A value of zero uses the standard lane.
+  /// @return fastTransferFeeBps The fee in basis points for fast transfers, or zero if standard transfer is used.
+  function _validateLockOrBurn(
+    Pool.LockOrBurnInV1 memory lockOrBurnIn,
+    uint16 finality
+  ) internal returns (uint16 fastTransferFeeBps) {
+    if (!isSupportedToken(lockOrBurnIn.localToken)) revert InvalidToken(lockOrBurnIn.localToken);
+    if (IRMN(i_rmnProxy).isCursed(bytes16(uint128(lockOrBurnIn.remoteChainSelector)))) revert CursedByRMN();
+    _checkAllowList(lockOrBurnIn.originalSender);
+
+    _onlyOnRamp(lockOrBurnIn.remoteChainSelector);
+    FastFinalityConfig storage finalityConfig = s_finalityConfig;
+    uint256 amount = lockOrBurnIn.amount;
+    if (finality != 0 && finalityConfig.finalityThreshold != 0) {
+      if (finality < finalityConfig.finalityThreshold) {
+        revert InvalidFinality(finality, finalityConfig.finalityThreshold);
+      }
+      if (amount > finalityConfig.maxAmountPerRequest) {
+        revert AmountExceedsMaxPerRequest(amount, finalityConfig.maxAmountPerRequest);
+      }
+
+      finalityConfig.outboundRateLimiterConfig[lockOrBurnIn.remoteChainSelector]._consume(
```

Comment:
Can we explicitly document in a comment that fast transfer rate limits and standard rate limits are completely independent?